saya/core/src/prover/atlantic/client.rs

```diff
@@ -324,10 +374,14 @@ impl AtlanticClient {
         Ok(response.text().await?)
     }
 
-    pub async fn get_trace(&self, id: &str) -> Result<Vec<u8>, ProverError> {
-        //TODO: now query returns the actual trace link. We need to change this to the actual trace link
-        //instead of the pie link being hardcoded
-        let url = format!("{}/{}/pie.cairo0.zip", ATLANTIC_S3_BASE, id);
+    pub async fn get_trace(
```

Comment:
Why not just implementing a function like `get_content` for the `MetadataUrls` structure?

Reply:
Like so we can remove `MetadataUrls` structure and just have some functions like `fetch_proof`, `fetch_trace` for the `AtlanticQueryResponse`.